src/main/scala/higherkindness/rules_scala/common/sbt-testing/Test.scala

```diff
@@ -29,7 +29,11 @@ object TestHelper {
     f: Runner => A
   ) = {
     val options =
-      if (framework.name == "specs2") Array("-ex", scopeAndTestName.replaceAll(".*::", "")) else Array.empty[String]
+      if (framework.name == "specs2") {
+        // Get rid of any extra context before the test name (specs2 doesn't recognize it)
```

Comment:
Nit: what is "extra content"?

Also: are any of the characters you are removing valid test name characters?

Third, can you walk me through the `\\Q` part of the regex? Do you need a corresponding `\E`? Are you interested in removing the `\` or does it still need to be a `\\` despite the raw string? Also, how does the capture group work with the `\\Q` right before the paren? Basically, I'm confused by the regex.

Reply:
Ah. This makes more sense after read the change in the next file.

I still have questions about `\\Q`, `\\E`, and `\\$`, though.

Reply:
Extra content: the test filter provided by intellij is the name of the class, a `#`, then whatever string is before the `should`, then a `::`, then whatever string is before the `in`. This removes the `should` stuff as well as the `::`, as as far as I can tell specs2 doesn't recognize those portions/that syntax.

Intellij adds a `\Q` and `\E` around everything after the `#` in the test filter. The code in the other file was removing that (idk why). I changed it so it wouldn't remove it, and then the `replaceAll` here needed to be updated to not touch the `\Q`. So the point of this regex is to change `\Qfoo.bar should::work properly\E$` to `\Qwork properly\E$`

Reply:
What benefit do we get from removing those things? I worry it might break other test frameworks beyond specs2, and I'm not certain what the benefit is.

Reply:
The benefit is that specs2 will run the tests :sweat_smile: If you don't remove the `foo.bar should::`, `specs2` just doesn't know what you're talking about: I think when filtering tests it only looks at the string label closest to the actual test, and none of the wrapping descriptions. The code was doing this before, it just didn't need to preserve the leading `\Q` as that had already been removed. So the existing code was changing `foo.bar should::work properly` to `work properly`, but would change `\Qfoo.bar should::work properly\E$` to `work properly\E$`.

The only way I changed the functionality was that the leading `\Q` will no longer be removed (and I added a comment explaining what the code was doing)

It shouldn't affect any other test frameworks because this logic is wrapped in a `if (framework.name == "specs2")`

Reply:
Awesome. Sounds great.